domain/modeldefaults/types.go

```diff
@@ -5,8 +5,41 @@ package modeldefaults
 
 import (
 	"reflect"
+
+	"github.com/juju/juju/environs/config"
 )
 
+// ModelDefaultAttributes represents a set of default values for
+// a given attribute. The values contain the schema default value,
+// any controller default value, and values set for cloud regions.
+type ModelDefaultAttributes map[string]AttributeDefaultValues
+
+// AttributeDefaultValues represents all the default values at each level for a given
+// setting.
+type AttributeDefaultValues struct {
+	// Default and Controller represent the values as set at those levels.
+	Default    any
+	Controller any
+	// Regions is a slice of Region representing the values as set in each
+	// region.
+	Regions []RegionDefaultValue
+}
+
+type CloudRegion struct {
```

Comment:
Question: Is there a better name for this struct? Reading the state interface I was very confused about the use.

Reply:
The old environs struct was called `CloudRegionSpec` which seemed a bit verbose so I shortened it to just `CloudRegion`

Reply:
Dropped to use separate service calls.